File: exercise/c#/Day01/Day01/Food.cs
namespace Day01
{
    public record Food(
        DateOnly ExpirationDate,
        bool ApprovedForConsumption,
        Guid? InspectorId)
    {
        public bool IsEdible(Func<DateOnly> now)
        {
            if (ExpirationDate.CompareTo(now()) > 0 &&
                ApprovedForConsumption &&
                InspectorId != null)
                return true;
            return false;
        }
    }
}

File: exercise/ts/day01/src/food.ts
type Maybe<T> = NonNullable<T> | null;

export class Food {
    constructor(
        public readonly expirationDate: Date,
        public readonly approvedForConsumption: boolean,
        public readonly inspectorId: Maybe<string>
    ) {
    }

    isEdible(now: () => Date): boolean {
        return this.expirationDate > now() &&
            this.approvedForConsumption &&
            this.inspectorId !== null;
    }
}